src/main/scala/it/unibo/authsim/library/user/builder/UserInformationBuilder.scala

```diff
@@ -54,8 +43,8 @@ class UserInformationBuilder extends Builder[Option[UserInformation]]:
    * @param cryptoInformation     CryptoInformation to use for the generation of the new userInformation
    * @returna                     UserInformationBuilder where the username field is setted with the provided value
    */
-  def withAlgorithmPolicy(cryptoInformation:CryptoInformation) : this.type =
-    this._cryptoInformation = cryptoInformation
+  def withAlgorithm(algorithm:CryptographicAlgorithm) : this.type =
+    val _algorithm=Some(algorithm)
     this
```

Comment:
This implementation could be substituted by `this.builderMethod` from `Builder` trait.
```suggestion
  def withAlgorithm(algorithm:CryptographicAlgorithm) = this.builderMethod[CryptographicAlgorithm](algorithm => this._algorithm = Some(algorithm))(algorithm)
```

Also, the current implementation do not update the variable `_algorithm` as you are creating a new `val` and do not change `this._algorithm`.